pkg/common/consumer/consumergroup_offsets_checker.go

```diff
@@ -0,0 +1,74 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/eventing-kafka/pkg/common/kafka/offset"
+)
+
+const (
+	OffsetCheckRetryTimeout  = 60 * time.Second
+	OffsetCheckRetryInterval = 5 * time.Second
+)
+
+// wrapper functions for the Sarama functions, to facilitate unit testing
+var newSaramaClient = sarama.NewClient
+var newSaramaClusterAdmin = sarama.NewClusterAdmin
+
+type ConsumerGroupOffsetsChecker interface {
+	WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error
+}
+
+type NoopConsumerGroupOffsetsChecker struct {
+}
+
+func (c *NoopConsumerGroupOffsetsChecker) WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error {
+	return nil
+}
+
+type KafkaConsumerGroupOffsetsChecker struct {
+}
+
+func (k *KafkaConsumerGroupOffsetsChecker) WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error {
+	logger.Infow("Checking if all offsets are initialized", zap.Any("topics", topics), zap.Any("groupID", groupID))
+
+	client, err := newSaramaClient(addrs, config)
+	if err != nil {
+		logger.Errorw("Unable to create Kafka client", zap.Error(err))
+		return err
+	}
+	defer client.Close()
+
+	clusterAdmin, err := newSaramaClusterAdmin(addrs, config)
+	if err != nil {
+		logger.Errorw("Unable to create Kafka cluster admin client", zap.Error(err))
+		return err
+	}
+	defer clusterAdmin.Close()
+
+	check := func() (bool, error) {
+		if initialized, err := offset.CheckIfAllOffsetsInitialized(client, clusterAdmin, topics, groupID); err == nil {
+			if initialized {
+				return true, nil
+			} else {
+				logger.Infow("Offsets not yet initialized, going to try again")
+				return false, nil
+			}
+		} else {
+			return true, fmt.Errorf("error checking if offsets are initialized. stopping trying. %w", err)
```

Comment:
```suggestion
			return false, fmt.Errorf("error checking if offsets are initialized. stopping trying. %w", err)
```

I think this is the correct behavior when you want to stop retrying in the error case. `PollUntil` will stop when there's an error

Reply:
I agree here too.  I'd also give a little comment above the  `	check := func() (bool, error)...` regarding this, and meanings

Reply:
```
PollUntil tries a condition func until it returns true, an error or stopCh is closed.
```

So, it won't matter if it is returning `true,err` or `false,err` as long as there's an error.

Reply:
There's a difference in the semantic meaning, returning `true, nil` means the check succeeded, stop retrying, returning `false, err` means the check failed with an error, stop retrying. 

Reading in the corde `true, err` is confusing knowing the semantic meaning behind it.

Reply:
Ok, changed it anyway